### PR TITLE
[SEDONA-746] Fix RS_Clip behavior

### PR DESCRIPTION
**Pull request by prantogg**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest/community/develop/)

## Is this PR related to a ticket?

- Yes, and the PR name follows the format `[SEDONA-XXX] my subject`.
- Yes, and the PR name follows the format `[GH-XXX] my subject`. Closes #<issue_number>

## What changes were proposed in this PR?

- Fixed Band Usage: Corrected issues related to improper handling of raster bands.
- Improved RS_Clip Algorithm: Resolved an issue where clipping was not performed correctly when the crop parameter was set to true.
- Fixed Misalignment in rasterizeGeomExtent: Addressed alignment issues to ensure accurate rasterization of geometry extents.
- Refactored Coordinate Transformations: Improved readability and fixed bugs in pixel-to-world and world-to-pixel transformations. Floating-point imprecisions were reduced by consistently using BigDecimal.
- Optimized Clipping Logic: Reduced redundant computations in the clip method by calculating offsets outside the loop.
- Enhanced Test Coverage:
  - Added dedicated tests for Rasterization.rasterizeGeomExtent.
  - Included edge case tests to ensure robustness.
  - Added tests for the band parameter in the clip method.

## How was this patch tested?

- Added new tests for edge cases
- Validated results using QGIS
- Passes existing tests for other functions that use rasterization (`RS_ZonalStats` and `RS_SetValues`)

## Did this PR include necessary documentation updates?

- No, this PR does not affect any public API so no need to change the documentation.

### CODE REVIEW

File: common/src/main/java/org/apache/sedona/common/raster/Rasterization.java [hunk 1]
```diff
@@ -574,17 +626,18 @@ private static Map<Double, TreeSet<Double>> computeScanlineIntersections(
           // calculating slope
           for (double y = yStart; y >= yEnd; y--) {
             double xIntercept = p1X; // Vertical line, xIntercept is constant
-            if (xIntercept < 0 || xIntercept >= params.writableRaster.getWidth()) {
+            if (xIntercept < 0 || xIntercept > params.writableRaster.getWidth()) {
```
[the remark this answers is not in the document]
Comment (prantogg) [reply common/src/main/java/org/apache/sedona/common/raster/Rasterization.java:629]: Having `>=` causes skipping of computing scanLineIntercepts treading the right hand side of the raster.
